src/header/common/cookie.rs

```diff
@@ -59,18 +77,82 @@ impl Header for Cookie {
     }
 }
 
+impl Cookie {
+    /// Create an empty Cookie header.
+    pub fn new() -> Cookie {
+        Cookie::with_capacity(0)
+    }
+
+    /// Create a Cookie header of a certain size.
+    pub fn with_capacity(capacity: usize) -> Cookie {
+        Cookie {
+            cookies: Vec::with_capacity(capacity),
+            index: VecMap::with_capacity(capacity),
+        }
+    }
+
+    /// Shrink the Cookie header internal elements to the currently used size.
+    pub fn shrink_to_fit(&mut self) {
+        self.cookies.shrink_to_fit();
+    }
+
+    /// Append a new cookie to the Cookie header.
+    pub fn push<T: Into<String>>(&mut self, name_tref: T, value_tref: T) {
+        let name = name_tref.into();
+        let value = value_tref.into();
+        self.cookies.push((name.clone(), value.clone()));
+        if self.index.get(&name) == None {
+            self.index.insert(name, value);
+        }
+    }
+
+
+    /// Get value of cookie from name. If duplicate names were pushed to the
+    /// Cookie header, this function will only return the first one.
+    pub fn get<T: Borrow<String>>(&self, name: T) -> Option<String> {
+        match self.index.get(name.borrow()) {
+            Some(value_ref) => Some((*value_ref).clone()),
+            None => None,
+        }
+    }
+
+
+    /// Clear the current Cookie, and add one with specified name and value.
+    pub fn set<T: Into<String>>(&mut self, name_tref: T, value_tref: T) {
```

Comment:
My understanding of the suggestion for this method was to clear the values of only this key. This would seem confusing:

```rust
let mut cookie = Cookie::new();
cookie.set("foo", "bar");
cookie.set("baz", "quux");
// where did my foo=bar go!
assert_eq!(cookie.to_string(), "baz=quux");
```